crypto/src/proofs/transparent.rs

```diff
@@ -587,6 +565,217 @@ impl TryFrom<transparent_proofs::SwapProof> for SwapProof {
     }
 }
 
+/// Transparent proof for swap creation.
+///
+/// Swaps create an output NFT encoding the swap data so they are most similar to Output operations.
+///
+/// This structure keeps track of the auxiliary (private) inputs.
+#[derive(Clone, Debug)]
+pub struct SwapProof {
+    // The diversified base for the destination address.
+    pub g_d: decaf377::Element,
+    // The transmission key for the destination address.
+    pub pk_d: ka::Public,
+    // The value of asset 1 in the swap.
+    pub value_t1: Value,
+    // The value of asset 2 in the swap.
+    pub value_t2: Value,
+    // The fee value associated with the swap.
+    // TODO: should always be of penumbra token type, maybe this should
+    // be uint64 or a hardcoded value?
+    pub value_fee: Value,
+    // The asset ID of the Swap NFT.
+    pub swap_nft_asset_id: asset::Id,
+    // The blinding factor used for generating the value commitment.
+    // TODO: can this be shared amongst t1/t2/fee or is this unsafe?
+    pub v_blinding: Fr,
```

Comment:
In general (and here), we need one blinding factor per revealed data -- since we have separate commitments to t1/t2/fee, we need one blinding factor for each.